### PR TITLE
Support static TFs for multi DOF joints in CurrentStateMonitor

### PR DESCRIPTION
**Pull request by pHaeusler**
### Description
The tfCallback method in CurrentStateMonitor disregards any TF lookups with a common time of zero. This makes it impossible to update a floating joint with a static TF because static TFs are published at time zero.

This fix adds support for static TF updates to floating joints in CurrentStateMonitor by also considering TFs with a common time of zero.

### CODE REVIEW

**Comment by v4hn** [comment]
I did not look at this yet, but Is this still relevant after https://github.com/ros-planning/moveit/pull/792 ?

**Comment by pHaeusler** [comment]
Yep. Currently any transforms with a common time of zero are not used to update floating joints, which precludes static transforms. This fix allows static transforms to update floating joints.

File: moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp [hunk 1]
```diff
@@ -402,68 +402,56 @@ void planning_scene_monitor::CurrentStateMonitor::tfCallback()
...
       }
-      joint_time_[joint] = latest_updates[i];
+      joint_time_[joint] = latest_common_time;
 
       Eigen::Affine3d eigen_transf;
       tf::transformTFToEigen(transf, eigen_transf);
 
-      double new_values[joint->getStateSpaceDimension()];
-      joint->computeVariablePositions(eigen_transf, new_values);
+      std::vector<double> new_values(joint->getStateSpaceDimension());
```
Comment (rhaschke) [inline moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp]: Why did you change `double[]` to `vector<double>`? The array is created on stack, while the vector requires heap allocation which is slower.

Comment (pHaeusler) [reply moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp]: When using [cpplint](https://github.com/cpplint/cpplint) I get this error
> Do not use variable-length arrays.  Use an appropriately named ('k' followed by CamelCase) compile-time constant for the size.  [runtime/arrays] [1]

Your comment about speed is valid though, so I will change it back.
